File: Combatant.js
    
public abstract class Combatant {
    protected String name;
    protected String element;
    protected String weakness;          
    protected int maximum_health;         
    protected int energy;             
    protected int maximum_energy;            
    protected int strength;           
    protected int intelligence;       
    protected int accuracy;           
    protected int speed;              
    protected int experience;                
    protected Story ongoing_story;
    
    public abstract void attack (Combatant target);
    
    public boolean is_Alive () {
        if (health > 0){
          return True;
         else{
          return False;
    }
    
    public void setName (String name) {
        this.name = name;
    }
    
    public void setLevel (double level) {
        this.level = level;
    }

    public void sethealth (double health) {
        this.health = health;
    }
    
    public void setmaxhealth (int Maxhealth) {
        this.maxhealth = maxhealth;
    }
    
    public void setenergy (int energy) {
        this.energy = energy;
    }
    
    public void setmaxenergy (int Maxenergy) {
        this.maxenergy = maxenergy;
    }
    
    public void setStrength (int strength) {
        this.strength = strength;
    }
    
    public void setResistance (int resistance) {
    this.resistance = resistance;
    }
    
    public void setIntelligence (int intelligence) {
        this.intelligence = intelligence;
    }
    
    public void setWisdom (int wisdom){
        this.wisdom = wisdom;
    }
    
    public void setAccuracy (int accuracy) {
        this.accuracy = accuracy;
    }
    
    public void setSpeed (int speed) {
        this.speed = speed;
    }
    
    public void setExp (int exp) {
        this.exp = exp;
    }
    
    public void setElement(String element){
        this.element = element;
    }
    
    public void setWeakness(String weakness){
        this.weakness = weakness;
    }
    
    public String getName () {
        return name;
    }
    
    public String getElement() {
        return element;

    }
    
    public String getWeakness() {
        return weakness;
    }
    
     
    public int getLevel () {
        return level;
    }
    
    public int getHealth () {
        return health;
    } 

    public int getHealthMax () {
        return healthMax;
    }
    
    public int getMana () {
        return mana;
    }

    public int getManaMax () {
        return manaMax;
    }
    
    public int getStrength () {
        return strength;
    }
    public int getIntelligence () {
        return intelligence;
    }
    
    public int getAccuracy () {
        return accuracy;
    }
    
    public int getSpeed () {
        return speed;
    }
    
    public int getExperience () {
        return experience;
    }
}

File: Combatant.java
    
public abstract class Combatant {
    protected String name;
    protected String element;
    protected String weakness;          
    protected int maximum_health;         
    protected int energy;             
    protected int maximum_energy;            
    protected int strength;          
    protected int speed;              
    protected int experience;                
    protected Story ongoing_story;
    
    public void attack (Combatant attacker);
    
    public boolean is_Alive () {
        int tester = health;
        if (tester > 0){
          return True;
         else{
          return False;
    }
    
    public void set_name (String id) {
        this.name = id;
    }
    
    public void set_level (int progress) {
        this.level = progress;
    }

    public void set_health (int welfare) {
        this.health = welfare;
    }
    
    public void set_max_health (int max_point) {
        this.maximum_health = max_point;
    }
    
    public void set_energy (float energy) {
        this.energy = energy;
    }
    
    public void set_max_energy (int energy_limit) {
        this.maximum_energy = energy_limit;
    }
    
    public void set_strength (int strength) {
        this.strength = strength;
    }
    
    public void resistance (int player_resistance) {
        this.resistance = player_resistance;
    }
    
    public void setIntelligence (int player_int) {
        this.intelligence = player_int;
    }
    
    public void set_wisdom (int wisdom){
        this.wisdom = wisdom;
    }
            
    public String get_name () {
        return name;
    }
    
    public String get_element() {
        return element;

    }
    
    public String get_weakness() {
        return weakness;
    }
    
    public int get_level () {
        return level;
    }
    
    public int get_health () {
        return health;
    } 

    public int get_maximum_health () {
        return maximum_health;
    }
        
    public int get_accuracy () {
        return accuracy;
    }
    
    public int get_speed () {
        return speed;
    }
    
    public int get_energy () {
        return energy;
    } 

    public int get_maximum_energy () {
        return maximum_energy;
    }

    public int get_experience () {
        return experience;
    }
}
